Shared/Supporting Files/Views/CategoryView.swift

```diff
@@ -24,20 +24,53 @@ struct CategoryView: View {
     /// The search query in the search bar.
     @Binding private(set) var query: String
     
+    /// The samples to display in the name section of the search list.
+    @State private var nameSearchResults: [Sample] = []
+    
+    /// The samples to display in the description section of the search list.
+    @State private var descriptionSearchResults: [Sample] = []
+    
+    /// The samples to display in the tags section of the search list.
+    @State private var tagsSearchResults: [Sample] = []
+    
     /// A Boolean value that indicates whether to present the about view.
     @State private var isAboutViewPresented = false
     
-    /// The samples to display in the search list. Searching adjusts this value.
-    private var displayedSamples: [Sample] {
-        searchSamples()
-    }
+    /// The names of the samples already found in a previous section.
+    @State var previousSearchResults: Set<String> = []
     
     var body: some View {
         Group {
             if !isSearching {
                 CategoryGridView(samples: samples)
             } else {
-                SampleListView(samples: displayedSamples)
+                // The search results list.
+                List {
+                    if !nameSearchResults.isEmpty {
+                        Section(header: Text("Name Results")) {
+                            SampleListView(samples: nameSearchResults, query: query)
+                        }
+                    }
+                    if !descriptionSearchResults.isEmpty {
+                        Section(header: Text("Description Results")) {
+                            SampleListView(samples: descriptionSearchResults, query: query)
+                        }
+                    }
+                    if !tagsSearchResults.isEmpty {
+                        Section(header: Text("Tag Results")) {
+                            SampleListView(samples: tagsSearchResults, query: query)
+                        }
+                    }
+                }
+                .onChange(of: query) { _ in
+                    previousSearchResults = []
```

Comment:
```suggestion
                    previousSearchResults.removeAll(keepingCapacity: true)
```
In this specific usecase, I think clearing the set while keeping capacity makes sense, as the set might be refilled immediately.